### ISSUE TITLE
[Docs] Add GPG signing requirement to contributor guide

### ISSUE THREAD

**Issue by nmsmith** [modularml/mojo#2871]
### Where is the problem?

https://github.com/nmsmith/mojo/blob/main/CONTRIBUTING.md

### What can we do better?

I recently submitted a PR to the `main` branch and according to @lattner, merging was blocked on my commit not being [GPG signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) (which I've now fixed). The [contributor guide](https://github.com/nmsmith/mojo/blob/main/CONTRIBUTING.md) doesn't mention anything about GPG signing. (This is a completely different concept to "signing off".) Does GPG signing need to be added to the contributor guide?

**Comment by gabrieldemarmiesse** [comment]
Maybe it's because you were targeting main instead of nightly? When targeting nightly it's never been a requirement.
I see that your PR was closed to retarget nightly. I'll just mention this issue there because it seems frequent that people make the mistake of targeting main: https://github.com/modularml/mojo/issues/2556 Feel free to +1 this issue since you made the mistake.

**Comment by nmsmith** [comment]
Yeah possibly this issue should be closed, if the maintainers know about GPG being a `main` branch thing and don’t intend to apply the same rule to `nightly`. Though it seems weird that the branches are configured differently.

**Comment by JoeLoser** [comment]
It's unintentional that the requirements are different between `main` and `nightly`.  There shouldn't be any GPG signing requirements for `main`, but like Gabriel said, it's partially moot, since we don't welcome contributions to the `main` branch.  It was a misunderstanding/oversight that Chris directly committed to the `main` branch.  All development happens on the `nightly` branch.

**Comment by scottamain** [comment]
I suspect this is obsolete now, please reopen if there’s still something we can improve.
